Foundation/JSONSerialization.swift

```diff
@@ -823,8 +852,21 @@ private struct JSONReader {
                 defer { intEndPointer.deallocate() }
                 let doubleEndPointer = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1)
                 defer { doubleEndPointer.deallocate() }
-                let intResult = strtol(startPointer, intEndPointer, 10)
+
+                let isNegative = startPointer.pointee == UInt8(ascii: "-")
+                let intResult: Int64
+                let uintResult: UInt64
+
+                if isNegative {
```

Comment:
Rather than repeating the result here, can we please instead advance past a `-` if present and always parse the remaining digits as `UInt64`? It will simplify the casting above and below to always work in terms of one type (`UInt64`) until you need to down cast it and check the range (esp. if we do have a `-`).

Reply:
Good idea. Will do